### ISSUE TITLE
Support firewall logs from Cortex Data Lake

### ISSUE THREAD

**Issue by rfaircloth-splunk** [PaloAltoNetworks/Splunk-Apps#162]
### Is your feature request related to a problem?

Customers using Splunk/Splunk Connect for Syslog have asked for support for cortex as a source in Splunk. 

### Describe the solution you'd like

On closer inspection we have realized the csv order of fields from cortext are not identical to pan sources preventing use of the existing addon source types. Update the addon to provide cortex:* version of pan:* with the correct order

### Describe alternatives you've considered

Considered have the customer reorder the fields in cortext via the UI but this seems impracticle at scale

### Additional context

Blocked use of cortext and Splunk

**Comment by btorresgil** [comment]
Logs from Cortex Data Lake have been supported for a long time using Log Forwarding in Cortex. However, a recent change to Log Forwarding made it so you can't use Splunk with Cortex if you have customized the filters or create new filters in your Log Forwarding Profile.

Important facts about this issue:

* The **only** filters in Cortex Log Forwarding that work with Splunk are 'migrated' filters.
* If you change migrated filters to regular filters, the logs will stop working with Splunk.
* The only way to change a filter to a migrated filter is to open a TAC case and request that your filters be converted to Migrated filters. TAC is aware of this issue and is converting many filters to migrated filters.
* This issue will be resolved in a future release of Log Forwarding for Cortex.

You can tell if you have a migrated filter in the Forwarding Profile. Any filters that work with Splunk will say ` - migrated` next to them. Here's an example of a profile with 4 migrated filters:

<img width="1305" alt="2021-02-10_10-03-24" src="https://user-images.githubusercontent.com/4164289/107551825-976a1180-6b87-11eb-86be-5fb1ede32e5b.png">

**Comment by btorresgil** [comment]
I've pinned this issue.

**Comment by MonkeyKa** [comment]
what is different about the migrated filter?  My team started recreating PA transforms and other KOs, but if a small tweak results in normal interpretation, I'll put a halt on that.

**Comment by btorresgil** [comment]
Migrated filter is any filter from before November that hasn't been changed. As soon as it's changed, it will no longer be migrated and will use the format you designate rather than the standard PANOS format.

If you would like to make a change but maintain the migrated filter for the standard PANOS format, you'll need to open a TAC case to make the change and restore the migrated status.

This will be resolved in a couple weeks when we release HEC log support for CDL and the Splunk Addon, which is a key value format so it won't have the issue.

**Comment by MonkeyKa** [comment]
I think that you have a specific understanding that I am not quite catching.  I think that you are saying that 

1. before November 2020 Cortex Data Lake sent logs in standard PANOS format and that the Splunk Add-on can read those logs
2. In November, the format changed, any user who did not want the change could submit a TAC case to revert to the old format
3. Soon CDL and the Splunk Addon will support logs via HEC.  These will be sent in key-value, so transforms will not be needed

Based on the Addon, I think that when we receive the migrated log format, we should set the sourcetype to pan:log for the CDL syslog data.  Is that correct?

Sorry if this all seems obvious to you, it may not be clear across the board.  We've been piloting PrismaAccess since the beginning of this year and our PaloAlto contacts did not tell us that there was a special format that we needed to request when forwarding logs to Splunk.  In fact a coworker asked support for more info and the response he got suggested to him that it was just supposed to work as is with the Addon.  (something that I already learned was not so, resulting in my team creating new sourcetypes prismaaccess:* and trying to map fields ourselves)

**Comment by btorresgil** [comment]
@MonkeyKa That's all correct, though for clarity I would rephrase point 2 as "In November, the format changed, but all existing users remained on the PANOS format unless they made a config change. Any new users or old users who made a config change got the new format. Any user who does not want the new format can submit a TAC case to revert to the PANOS format.

Thankfully this will all be resolved by HEC support which is coming very soon. We're wrapping up the Splunk Add-on with HEC support now so all the CIM mappings and our Splunk App dashboards will work regardless if you use PANOS syslogs or CDL with HEC.
